ecl/hql/hqlrepository.cpp

```diff
@@ -823,29 +824,79 @@ unsigned EclRepositoryManager::runGitCommand(StringBuffer * output, const char *
     if (!output)
         output= &tempOutput;
 
+    Owned<IFile> extractedKey;
     EnvironmentVector env;
-    const char * username = getenv("HPCC_GIT_USERNAME");
     //If fetching from git and the username is specified then use the script file to provide the username/password
-    if (needCredentials && username)
+    if (needCredentials)
     {
-        StringBuffer scriptPath;
-        getPackageFolder(scriptPath);
-        addPathSepChar(scriptPath).append("bin/hpccaskpass.sh");
-        env.emplace_back("GIT_ASKPASS", scriptPath);
+        //If the username is supplied, then get the secret and write it to a temporary location.
+        bool useScript = false;
+        if (options.gitUser.length())
+        {
+            env.emplace_back("HPCC_GIT_USERNAME", options.gitUser.str());
+
+            if (!options.gitPasswordPath.isEmpty())
```

Comment:
@shamser this is the bulk of the implementation of HPCC-26907.  It may be all that is needed - but needs testing in baremetal and k8s.